### PR TITLE
Add xml file for running ND reconstruction using LArSED input.

### PR DESCRIPTION
**Pull request by yangtj207**
Add a xml file for running ND reconstruction using LArSED input and with LArHierarchyValidation.

### CODE REVIEW

File: settings/PandoraSettings_LArSED.xml [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+        <InputMCParticleListName>Input</InputMCParticleListName>
+    </algorithm>
+
+    <algorithm type = "LArMCParticleMonitoring">
+        <CaloHitListName>CaloHitList2D</CaloHitListName>
+        <MCParticleListName>Input</MCParticleListName>
+        <MinHitsForDisplay>0</MinHitsForDisplay>
+    </algorithm>
+
+<!--
```
Comment (jback08) [inline settings/PandoraSettings_LArSED.xml:65]: I would like this xml file to only use the hierarchy tools, so we should remove this `LArNeutrinoEventValidation` code block, even though it is commented out.

File: settings/PandoraSettings_LArSED.xml [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+      <CaloHitListName>CaloHitList2D</CaloHitListName>
+        <PfoListName>SliceParticles</PfoListName>
+        <VisualizePFO>true</VisualizePFO>
+    </algorithm>
+-->
+
+<algorithm type = "LArHierarchyValidation">
+   <ValidateEvent>true</ValidateEvent>
+   <WriteTree>true</WriteTree>
+   <FileName>EventValidation.root</FileName>
```
Comment (jback08) [inline settings/PandoraSettings_LArSED.xml:103]: Perhaps we should use `EventHierarchy.root` and `EventHierarchy` for the tree name? Then we won't potentially confuse new `hierarchy` output files with previous `validation` ones.

File: settings/PandoraSettings_LArSED.xml [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+   <TreeName>EventValidation</TreeName>
+   <FoldToPrimaries>true</FoldToPrimaries>
+</algorithm>
+
+<algorithm type = "LArHierarchyValidation">
+   <ValidateMC>true</ValidateMC>
+   <WriteTree>true</WriteTree>
+   <FileName>MCValidation.root</FileName>
```
Comment (jback08) [inline settings/PandoraSettings_LArSED.xml:111]: Similarly, could we use `MCHierarchy.root` and `MCHierarchy` for the tree name?

File: settings/PandoraSettings_LArSED.xml [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+    <algorithm type = "LArVisualMonitoring">
+        <MCParticleListNames>Input</MCParticleListNames> -->
+        <!--Suppress neutrons (PDG=2112) with E < 0.95 GeV-->
+<!--        <SuppressMCParticles>2112:0.95</SuppressMCParticles>
+	<CaloHitListNames>CaloHitListU CaloHitListV CaloHitListW</CaloHitListNames>
+        <ShowDetector>true</ShowDetector>
+    </algorithm>
+-->
+
+<!--
```
Comment (jback08) [inline settings/PandoraSettings_LArSED.xml:27]: Perhaps add a text comment here such as `enable this for event visualisation` ?

File: settings/PandoraSettings_LArSED.xml [hunk 5]
```diff
@@ -0,0 +1,121 @@
+<pandora>
+    <!-- GLOBAL SETTINGS -->
+    <IsMonitoringEnabled>true</IsMonitoringEnabled>
+    <ShouldDisplayAlgorithmInfo>false</ShouldDisplayAlgorithmInfo>
+    <SingleHitTypeClusteringMode>true</SingleHitTypeClusteringMode>
+
+    <!-- ALGORITHM SETTINGS -->
+    <!--algorithm type = "LArEventReading"/-->
```
Comment (jback08) [inline settings/PandoraSettings_LArSED.xml:8]: This line can be removed (`LArEventReading`) ?

File: settings/PandoraSettings_LArSED.xml [hunk 6]
```diff
@@ -0,0 +1,121 @@
...
+        <PfoListName>RecreatedPfos</PfoListName>
+        <UseTrueNeutrinosOnly>true</UseTrueNeutrinosOnly>
+        <PrintAllToScreen>true</PrintAllToScreen>
+        <PrintMatchingToScreen>true</PrintMatchingToScreen>
+        <WriteToTree>true</WriteToTree>
+        <OutputTree>Validation</OutputTree>
+        <OutputFile>Validation.root</OutputFile>
+    </algorithm>
+-->
+<!--
```
Comment (jback08) [inline settings/PandoraSettings_LArSED.xml:78]: Should we replace this with another call to `LArHierarchyMonitoring`? Also add a comment text: "enable to see PFOs"

File: settings/PandoraSettings_LArSED.xml [hunk 7]
```diff
@@ -0,0 +1,121 @@
...
+    <algorithm type = "LArPreProcessing">
+        <OutputCaloHitListNameU>CaloHitListU</OutputCaloHitListNameU>
+        <OutputCaloHitListNameV>CaloHitListV</OutputCaloHitListNameV>
+        <OutputCaloHitListNameW>CaloHitListW</OutputCaloHitListNameW>
+        <FilteredCaloHitListName>CaloHitList2D</FilteredCaloHitListName>
+        <CurrentCaloHitListReplacement>CaloHitList2D</CurrentCaloHitListReplacement>
+    </algorithm>
+
+<!--
+    <algorithm type = "LArVisualMonitoring">
```
Comment (jback08) [inline settings/PandoraSettings_LArSED.xml:18]: Is `LArVisualMonitoring` now replaced by the use of `LArHierarchyMonitoring`? If so, can we remove this?

File: settings/PandoraSettings_LArSED.xml [hunk 8]
```diff
@@ -0,0 +1,121 @@
...
+-->
+<!--
+    <algorithm type = "LArVisualMonitoring">
+        <ShowCurrentPfos>true</ShowCurrentPfos>
+        <ShowDetector>true</ShowDetector>
+    </algorithm>
+-->
+
+<!--
+    <algorithm type = "LArVisualParticleMonitoring">
```
Comment (jback08) [inline settings/PandoraSettings_LArSED.xml:86]: Do we have hierarchy equivalents for these visualisation algorithms?

**Review by jback08 — CHANGES_REQUESTED** [review]
Thanks for this, Tingjun. 

I would like this file to be the one that enables hierarchy tools for both SED and EDepSim data formats, and so this should be renamed to something like `PandoraSettings_LArRecoNDHierarchy.xml`, since the xml file works for any data format.

I have some comments and questions below about the various algorithm calls.

**Comment by yangtj207** [comment]
Hi @jback08 I've addressed most of your comments. I simply removed all commented out blocks since I figured we could add them back if needed in the future.

**Comment by jback08** [comment]
Thanks for the updates, @yangtj207 . However, could you please add back in all of the event display `hierarchy` settings, although leave them commented out, so that the user just has to uncomment them to be able to use the display, i.e. what complete set of xml commands did you have in order to create the plots shown in the DUNE-ND slack channel?

I just wanted to remove all instances of the `validation` settings.

**Comment by jback08** [comment]
OK, I was misunderstanding what was required for the visualisation steps, and this simplifies the xml file. Thanks, Tingjun.

**Pull request merged**